src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptCompletionProvider.cs

```diff
@@ -15,12 +15,13 @@ internal abstract class VSTypeScriptCompletionProvider : CompletionProvider
         public sealed override bool ShouldTriggerCompletion(SourceText text, int caretPosition, CompletionTrigger trigger, OptionSet options)
         {
             Debug.Fail("For backwards API compat only, should not be called");
-            return ShouldTriggerCompletionImpl(text, caretPosition, trigger, CompletionOptions.Default.TriggerOnTypingLetters);
+            var defaultOptions = CompletionOptions.Default;
+            return ShouldTriggerCompletionImpl(text, caretPosition, trigger, defaultOptions.TriggerOnTypingLetters, defaultOptions.ExpandedCompletionBehavior);
         }
 
         internal sealed override bool ShouldTriggerCompletion(HostLanguageServices languageServices, SourceText text, int caretPosition, CompletionTrigger trigger, CompletionOptions options, OptionSet passThroughOptions)
-            => ShouldTriggerCompletionImpl(text, caretPosition, trigger, options.TriggerOnTypingLetters);
+            => ShouldTriggerCompletionImpl(text, caretPosition, trigger, options.TriggerOnTypingLetters, options.ExpandedCompletionBehavior);
 
-        protected abstract bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters);
+        protected abstract bool ShouldTriggerCompletionImpl(SourceText text, int caretPosition, CompletionTrigger trigger, bool triggerOnTypingLetters, ExpandedCompletionMode completionMode);
```

Comment:
>ExpandedCompletionMode [](http://example.com/codeflow?start=152&length=22)

`ExpandedCompletionMode` is an internal Roslyn type and shouldn't flow into TypeScript. You can either define `VSTypeScriptExpandedCompletionMode` enum with the same values and map them, or pass a boolean if you don't need all the different cases.

Reply:
Noted. Resolving this since we didn't end up changing the method signature at all.